docs/source/installation/configuration.md

```diff
@@ -161,14 +161,30 @@ load_utilities:
 
 ## Middleware
 
-`guillotina` is built on `asgi` which provides support for middleware.
+`guillotina` support two types of middlewares:
+
+1. ASGI middlewares: run before the guillotina asgi application and can be used to
+  catch errors or to record timing metrics
+2. Guillotina (aiohttp-like) middlewares: run after the router and before executing the view
+
 You can provide an array of dotted names to use for your application.
 
 ```yaml
+asgi_middlewares:
```

Comment:
Should we document that the order defined here matters?

For example, if we provide error handling middleware, it needs to always be last.

Also, have you thought about how to pass configuration to the middleware?

Reply:
> For example, if we provide error handling middleware, it needs to always be last.

I'd say the ErrorMiddleware will be added dynamically

> Also, have you thought about how to pass configuration to the middleware?

The middleware can read the `app_settings` directly.